packages/nimble-components/src/label-provider/base/index.ts

```diff
@@ -0,0 +1,70 @@
+import { DesignToken, FoundationElement } from '@microsoft/fast-foundation';
+import { Notifier, Observable, type Subscriber } from '@microsoft/fast-element';
+import { themeProviderTag } from '../../theme-provider';
+
+/**
+ * Base class for label providers
+ */
+export abstract class LabelProviderBase
+    extends FoundationElement
+    implements Subscriber {
+    protected abstract readonly supportedLabels: {
```

Comment:
[Played with it a bit](https://www.typescriptlang.org/play?#code/C4TwDgpgBAIhDOBLA5gOwCoHsDWFXwDFMAnAHgHkAjAKwgGNh0A+KAXigG8AoKXqAbVwgoiVFCGYAZlCq0G6ALoAuKPGDFRyANxcAvly4BDSmuKGGUOgBtD8eFAAyxiFYAKxTADdEAEwjEAIVsIUgBlAFcwMBJgCB8nShd7CAAPWNQfew5BCBAVU01lKFRwgFtE4igAegAqSwALQ1RkaGBMKD8kNDbcVBqq3RZuPihjU3NgKFKQBKSVCKiYuNmreB19LjpMfEnplft2Yb4bRKsARhUAVgAaHmPnKwAmFQBmPVH7LZ2dTZs7KAAsiAAMJ-ZJpPCZRwPdxeXz+ILwEKgSBSKYzB7wFiIUpgKwQUp4YD2OBdDA4PCEEikFEQNF7TFDO68YgQQw+bZWYQM04HdH7HQjE4uM5sKAAcnqLismHFgvup0eYvFAHcSFYfHKDPogA) and I think that can make the types a bit stronger by passing the supported labels as a generic but if not it's fine since it's internal stuff.

Reply:
Updated